lib/hexpm/accounts/web_auth.ex

```diff
@@ -0,0 +1,113 @@
+defmodule Hexpm.Accounts.WebAuth do
+  use Hexpm.Context
+
+  @moduledoc false
+
+  # A pool for storing and validating web auth requests.
+
+  alias Hexpm.Accounts.WebAuthRequest
+  alias Hexpm.Accounts.Keys
+
+  # `device_code` refers to the code assigned to a client to identify it
+  # `user_code` refers to the code the user enters to authorize a client
+  # `scope` refers to the permissions granted to a key
+  # `scopes` refers to the list of scopes that are allowed in a web auth request
+  # `key_permission` is the default permissions given to generate a hex api key
+  # `key_params` is the default params to generate a hex api key
+
+  @key_permission %{domain: "api", resource: nil}
+  @key_params %{name: nil, permissions: [@key_permission]}
+
+  @doc """
+  Adds a web auth request to the pool and returns the response.
+
+  ## Params
+
+    - `scope` - Scope of the key to be generated. One of read and write.
+  """
+
+  def get_code(params) do
+    device_code = "foo"
+    user_code = "bar"
+
+    params = %{
+      device_code: device_code,
+      user_code: user_code,
+      scope: params
```

Comment:
sorry, I didn’t mean to simply rename scope to params. What I had in mind was params be a map, scope be one of its keys.